### DISCUSSION TITLE
Credo Mediator

### DISCUSSION THREAD

**Discussion by TimoGlastra** [openwallet-foundation/credo-ts#2136] [Architecture & Design]
Discussion to coordinate the creation of a Credo Mediator.

There are currently four different initiatives:
- 2060 message pickup: https://github.com/2060-io/message-pickup-repository
- animo mediator: https://github.com/animo/animo-mediator
- credebl mediator (forked from animo-mediator, but extended with more features): https://github.com/credebl/mediator-agent
- BC mediator: ??

Steps:
- [ ] Create a new credo-mediator repository under OWF (@TimoGlastra)
	- [ ] 	set up as monorepo so we can also add the packages for the client that a credo wallet can leverage to connect
- [ ] Add a base mediator implementation based on one of the existing initiatves
	- [ ]  Start with Credebl (they will transfer to Animo, animo will transfer to OWF)
- [ ] Move over interesting parts from the other initiatives (socketdock, redis, etc..)
	- [ ] Plugins from 2060.io's implementation (redis / postgres)
	- [ ] Socketdock from credebls 
- [ ] Add docker container publishing to GH container registry
	- [ ] Something BC Gov can pick up help with?
- [ ] Add K8S deployment files
	- [ ] Something BC Gov can pick up help with?

I think we should use the message pickup modules from 2060.io, and add the socket dock implementation from Credbl to this.

Ideally it becomes a monorepo where we add a server (mediator agent) and we add the subpackages for the socketdock etc..  I think we can make the socketdock optional (if you don't need HA it's not needed).

cc @jleach @sairanjit @genaris @GHkrishna

**Comment by jleach** [comment]
> Add docker container publishing to GH container registry
 Something BC Gov can pick up help with?
 Add K8S deployment files
 Something BC Gov can pick up help with?

@TimoGlastra I mostly have these last two points done. Once we have an OWF repo I can fix the CICD workflow to build the image and then touch up the helm deployment charts for k8s which I've been dumping [here](https://github.com/fullboar/mediator-agent) for now.

EDIT: And make some deployment docs.

**Comment by swcurran** [comment]
I would recommend that once you pick the repo to use, that you move it (with Ry’s help) into OWF, we’ll add maintainers and you can go from there.  Are we agreed to put it under the DIDComm Mediator project, and call it `didcomm-mediator-service-credo`?  If so, I’ll get the existing repo renamed to `didcomm-mediator-service-acapy`.  Happy to drop the `-service` if we want :-).

Let me know.

**Reply by jleach** [reply]
I think we agreed this morning that @TimoGlastra will tidy up the Animo repo and transfer it into OWF. That will be the starting point. I can merge my PR in there which is based of [credebl/mediator-agent](https://github.com/credebl/mediator-agent) so we'll get their changes updating to Credo v0.5.x.

Original maintainers are @TimoGlastra + everyone CCed in his original post I think. That's Animo, Credeble, 2060.io, and BC.

**Comment by sairanjit** [comment]
@TimoGlastra I have added a PR in Animo Mediator https://github.com/animo/animo-mediator/pull/28 with existing features

**Comment by TimoGlastra** [comment]
We now have: https://github.com/openwallet-foundation/didcomm-mediator-credo, so this can be closed
